From d22cc5cbbbd1d8e9c801e7851c9a4378ed61bcfa Mon Sep 17 00:00:00 2001
From: Yuly Novikov <ynovikov@chromium.org>
Date: Thu, 4 Jan 2018 17:15:57 -0500
Subject: Vulkan: Use correct Linux link flags for layers.

From third_party/vulkan-validation-layers/src/layers/CMakeLists.txt
Otherwise, there is a conflict between global functions with same name
from different layers and the loader.
Conflict is resolved by -Bsymbolic.

BUG=angleproject:2296

Change-Id: I40859326626c3865ba346c6e817b7c512735054e
Reviewed-on: https://chromium-review.googlesource.com/850727
Commit-Queue: Yuly Novikov <ynovikov@chromium.org>
Reviewed-by: Jamie Madill <jmadill@chromium.org>

diff --git a/src/vulkan_support/BUILD.gn b/src/vulkan_support/BUILD.gn
index 0efc586a7..fe618262c 100644
--- a/src/vulkan_support/BUILD.gn
+++ b/src/vulkan_support/BUILD.gn
@@ -863,6 +863,9 @@ foreach(layer_info, layers) {
     if (is_win) {
       sources += [ "$vulkan_layers_dir/layers/VkLayer_$name.def" ]
     }
+    if (is_linux) {
+      ldflags =  [ "-Wl,-Bsymbolic,--exclude-libs,ALL" ]
+    }
   }
 }
 
-- 
2.14.1

